code/modules/mob/living/simple_animal/eldritch_demons.dm

```diff
@@ -0,0 +1,276 @@
+	minbodytemp = 0
+	maxbodytemp = INFINITY
+	healable = 0
+	faction = list("e_cult")
+	movement_type = GROUND
+	pressure_resistance = 100
+	del_on_death = TRUE
+	deathmessage = "implodes into itself"
+
+/mob/living/simple_animal/hostile/eldritch/raw_prophet
+	name = "Raw Prophet"
+	real_name = "Raw Prophet"
+	desc = "Abomination made from severed limbs."
+	icon_state = "raw_prophet"
+	status_flags = CANPUSH
+	icon_living = "raw_prophet"
+	melee_damage_lower = 5
+	melee_damage_upper = 10
+	maxHealth = 75
+	health = 75
+	sight = SEE_MOBS|SEE_OBJS|SEE_TURFS
+
+/mob/living/simple_animal/hostile/eldritch/raw_prophet/Initialize()
+	. = ..()
+	AddSpell(new /obj/effect/proc_holder/spell/targeted/ethereal_jaunt/shift/ash/long)
+	AddSpell(new /obj/effect/proc_holder/spell/targeted/telepathy/eldritch)
+
+/mob/living/simple_animal/hostile/eldritch/raw_prophet/Login()
+	. = ..()
+	client.change_view(11)
+
+/mob/living/simple_animal/hostile/eldritch/armsy
+	name = "Terror of the night"
+	real_name = "Armsy"
+	desc = "Abomination made from severed limbs."
+	icon_state = "armsy_start"
+	icon_living = "armsy_start"
+	maxHealth = 200
+	health = 200
+	melee_damage_lower = 10
+	melee_damage_upper = 15
+	var/mob/living/simple_animal/hostile/eldritch/armsy/back
+	var/mob/living/simple_animal/hostile/eldritch/armsy/front
+	var/oldloc
+	var/allow_pulling = FALSE
+	var/stacks_to_grow = 5
+	var/current_stacks = 0
+
+/mob/living/simple_animal/hostile/eldritch/armsy/New(spawn_more = TRUE,len = 6)
+	. = ..()
+	oldloc = loc
+	if(!spawn_more)
+		return
+	allow_pulling = TRUE
+	var/mob/living/simple_animal/hostile/eldritch/armsy/next
+	var/mob/living/simple_animal/hostile/eldritch/armsy/prev
+	var/mob/living/simple_animal/hostile/eldritch/armsy/current
+	for(var/i = 0, i <= len,i++)
+		prev = current
+		//i tried using switch, but byond is really fucky and it didnt work as intended. Im sorry
+		if(i == 0)
+			current = new type(drop_location(),spawn_more = FALSE)
+			current.icon_state = "armsy_mid"
+			current.icon_living = "armsy_mid"
+			current.front = src
+			current.AIStatus = AI_OFF
+			back = current
+		if(i > 0 && i < len)
+			current = new type(drop_location(),spawn_more = FALSE)
+			prev.back = current
+			prev.icon_state = "armsy_mid"
+			prev.icon_living = "armsy_mid"
+			prev.front = next
+			prev.AIStatus = AI_OFF
+		if(i == len)
+			prev.icon_state = "armsy_end"
+			prev.icon_living = "armsy_end"
+			prev.front = next
+			prev.AIStatus = AI_OFF
+		next = prev
+
+/mob/living/simple_animal/hostile/eldritch/armsy/Moved()
+
+	if(pulledby && !allow_pulling)
+		pulledby.stop_pulling()
+		loc = oldloc
+
+	if(back && back.loc != oldloc)
+		for(var/i = 0, i < max(get_dist(loc,back.loc),10),i++)
+			step_towards(back,oldloc)
+			if(loc == back.loc)
+				break
+		back.loc = oldloc//just in case
+
+	if(front && loc != front.oldloc)
+		for(var/i = 0, i < max(get_dist(loc,front.loc),10),i++)
```

Comment:
Use the list loop if you are not manipulating the index inside the loop.
In this case it'd be `for(var/i in 1 to max(get_dist(loc, front.loc),10))`